pandas/core/arrays/masked.py

```diff
@@ -417,7 +417,7 @@ def isin(self, values) -> BooleanArray:  # type: ignore[override]
             # see https://github.com/pandas-dev/pandas/pull/38379 for some discussion
             result[self._mask] = values_have_NA
 
-        mask = np.zeros_like(self, dtype=bool)
+        mask = np.zeros(self._data.shape, dtype=bool)
```

Comment:
Why not using ``np.zeros_like(self._data.shape, dtype=bool)``?

```
%timeit np.zeros_like(df._data.shape, dtype=bool)
4.11 µs ± 98.7 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
%timeit np.zeros(df._data.shape, dtype=bool)
20.6 µs ± 360 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```

Reply:
DataFrame had shape ``(10_000, 100)``

Reply:
Hmm I can't recreate this, can you please give a full example? From what you have above, if you pass a shape into `np.zeros_like` it is faster because it is just allocating 2 values, eg
```
In [8]: a = np.zeros_like(arr.shape, dtype=bool)

In [9]: a
Out[9]: array([False, False])
```

It makes sense for `zeros_like` to be slower based on impl:
```
res = empty_like(a, dtype=dtype, order=order, subok=subok, shape=shape)
# needed instead of a 0 to get same result as zeros for for string dtypes
z = zeros(1, dtype=res.dtype)
multiarray.copyto(res, z, casting='unsafe')
```
So essentially a `malloc` then `memset` instead of just the `calloc` which `zeros` will do.

Reply:
Hm good point, I viewed np.zeros twice instead of each once. Forget my comment